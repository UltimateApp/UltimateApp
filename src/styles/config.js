export default {
  swipeConfig: {
    velocityThreshold: 0.2,
    directionalOffsetThreshold: 80,
  },
};
