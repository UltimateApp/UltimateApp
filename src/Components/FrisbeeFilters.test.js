import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { createDrill } from '../Fixtures/TestFixtures';
import { render, fireEvent } from '@testing-library/react-native';
import { Levels, FrisbeeGoals, DrillTypes } from '../Fixtures/config';

import ConnectedFrisbeeFilters, { FrisbeeFilters } from './FrisbeeFilters';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<FrisbeeFilters />', () => {
  const beginnerDrill = createDrill({
    id: 1,
    type: DrillTypes.FRISBEE,
    goals: [FrisbeeGoals.HANDLING],
    level: Levels.BEGINNER,
  });
  const intermediateDrill = createDrill({
    id: 2,
    type: DrillTypes.FRISBEE,
    goals: [FrisbeeGoals.HANDLING],
    level: Levels.INTERMEDIATE,
  });
  const advancedDrill = createDrill({
    id: 3,
    type: DrillTypes.FRISBEE,
    goals: [FrisbeeGoals.HANDLING],
    level: Levels.ADVANCED,
  });

  it('renders correctly', () => {
    const route = {
      params: {
        initialData: [beginnerDrill, advancedDrill],
      },
    };
    const navigation = { setOptions: jest.fn(), navigate: jest.fn() };
    const tree = renderer.create(<FrisbeeFilters route={route} navigation={navigation} favoriteDrills={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('filtering', () => {
    it('filters drills by level', async () => {
      const drills = [beginnerDrill, intermediateDrill, advancedDrill];
      const navigate = jest.fn();

      const DummyScreen = props => null;
      const Stack = createStackNavigator();

      const mockStore = configureMockStore()({
        favoriteDrills: [],
      });

      const { getByText, getByTestId } = render(
        <Provider store={mockStore}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="FrisbeeFilters"
                component={ConnectedFrisbeeFilters}
                initialParams={{
                  initialData: drills,
                  previousScreen: 'DrillListPage',
                  previousType: DrillTypes.FRISBEE,
                }}
                listeners={({ navigation }) => ({
                  transitionStart: e => {
                    navigation.navigate = navigate;
                  },
                })}
              />
              <Stack.Screen name="DrillListPage" component={DummyScreen} />
            </Stack.Navigator>
          </NavigationContainer>
          ,
        </Provider>,
      );

      expect(getByText('3 drills available')).toBeDefined();

      await fireEvent.press(getByText(Levels.BEGINNER));

      expect(getByText('1 drill available')).toBeDefined();

      await fireEvent.press(getByText(Levels.INTERMEDIATE));

      expect(getByText('2 drills available')).toBeDefined();

      await fireEvent.press(getByText(Levels.BEGINNER));

      expect(getByText('1 drill available')).toBeDefined();

      await fireEvent.press(getByText(Levels.INTERMEDIATE));

      expect(getByText('3 drills available')).toBeDefined();

      await fireEvent.press(getByText(Levels.ADVANCED));

      await fireEvent.press(getByTestId('validateButton'));

      expect(navigate).toBeCalledWith('DrillListPage', {
        type: DrillTypes.FRISBEE,
        currentFilters: {
          selectedFavorites: false,
          selectedLevels: [Levels.ADVANCED],
          selectedGoals: [],
          numberOfPlayers: undefined,
          displayedDrills: [advancedDrill],
        },
      });
    });

    it('filters drills by goal', async () => {
      const defenseDrill = createDrill({ id: 1, goals: [FrisbeeGoals.DEFENSE] });
      const handlingDrill = createDrill({ id: 2, goals: [FrisbeeGoals.HANDLING] });
      const throwingDrill = createDrill({ id: 3, goals: [FrisbeeGoals.THROWING] });
      const handlingDefenseDrill = createDrill({ id: 4, goals: [FrisbeeGoals.DEFENSE, FrisbeeGoals.HANDLING] });
      const drills = [defenseDrill, handlingDrill, throwingDrill, handlingDefenseDrill];
      const navigate = jest.fn();

      const DummyScreen = props => null;
      const Stack = createStackNavigator();

      const mockStore = configureMockStore()({
        favoriteDrills: [],
      });

      const { getByText, getByTestId } = render(
        <Provider store={mockStore}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="FrisbeeFilters"
                component={ConnectedFrisbeeFilters}
                initialParams={{
                  initialData: drills,
                  previousScreen: 'DrillListPage',
                  previousType: DrillTypes.FRISBEE,
                }}
                listeners={({ navigation }) => ({
                  transitionStart: e => {
                    navigation.navigate = navigate;
                  },
                })}
              />
              <Stack.Screen name="DrillListPage" component={DummyScreen} />
            </Stack.Navigator>
          </NavigationContainer>
          ,
        </Provider>,
      );

      expect(getByText('4 drills available')).toBeDefined();

      await fireEvent.press(getByText(FrisbeeGoals.DEFENSE));

      expect(getByText('2 drills available')).toBeDefined();

      await fireEvent.press(getByText(FrisbeeGoals.THROWING));

      expect(getByText('3 drills available')).toBeDefined();

      await fireEvent.press(getByText(FrisbeeGoals.DEFENSE));

      expect(getByText('1 drill available')).toBeDefined();

      await fireEvent.press(getByText(FrisbeeGoals.THROWING));

      expect(getByText('4 drills available')).toBeDefined();

      await fireEvent.press(getByText(FrisbeeGoals.HANDLING));

      expect(getByText('2 drills available')).toBeDefined();

      await fireEvent.press(getByTestId('validateButton'));

      expect(navigate).toBeCalledWith('DrillListPage', {
        type: DrillTypes.FRISBEE,
        currentFilters: {
          selectedFavorites: false,
          selectedLevels: [],
          selectedGoals: [FrisbeeGoals.HANDLING],
          numberOfPlayers: undefined,
          displayedDrills: [handlingDrill, handlingDefenseDrill],
        },
      });
    });

    it('filters drills by number of players', async () => {
      const onePersonDrill = createDrill({ id: 1, minimalPlayersNumber: 1 });
      const twoPeopleDrill = createDrill({ id: 2, minimalPlayersNumber: 2 });
      const tenPeopleDrill = createDrill({ id: 3, minimalPlayersNumber: 10 });
      const drills = [onePersonDrill, twoPeopleDrill, tenPeopleDrill];
      const navigate = jest.fn();

      const DummyScreen = props => null;
      const Stack = createStackNavigator();

      const mockStore = configureMockStore()({
        favoriteDrills: [],
      });

      const { getByText, getByTestId } = render(
        <Provider store={mockStore}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="FrisbeeFilters"
                component={ConnectedFrisbeeFilters}
                initialParams={{
                  initialData: drills,
                  previousScreen: 'DrillListPage',
                  previousType: DrillTypes.FRISBEE,
                }}
                listeners={({ navigation }) => ({
                  transitionStart: e => {
                    navigation.navigate = navigate;
                  },
                })}
              />
              <Stack.Screen name="DrillListPage" component={DummyScreen} />
            </Stack.Navigator>
          </NavigationContainer>
          ,
        </Provider>,
      );

      expect(getByText('Number of players: -')).toBeDefined();
      expect(getByText('3 drills available')).toBeDefined();

      await fireEvent(getByTestId('numberOfPlayersSlider'), 'valueChange', 5);

      expect(getByText('Number of players: 5')).toBeDefined();
      expect(getByText('2 drills available')).toBeDefined();

      await fireEvent.press(getByTestId('validateButton'));

      expect(navigate).toBeCalledWith('DrillListPage', {
        type: DrillTypes.FRISBEE,
        currentFilters: {
          selectedFavorites: false,
          selectedLevels: [],
          selectedGoals: [],
          numberOfPlayers: 5,
          displayedDrills: [onePersonDrill, twoPeopleDrill],
        },
      });
    });

    it('filters favorite drills', async () => {
      const drills = [beginnerDrill, intermediateDrill, advancedDrill];
      const navigate = jest.fn();

      const DummyScreen = props => null;
      const Stack = createStackNavigator();

      const mockstore = configureMockStore()({
        favoriteDrills: [intermediateDrill],
      });

      const { getByText, getByTestId } = render(
        <Provider store={mockstore}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="FrisbeeFilters"
                component={ConnectedFrisbeeFilters}
                initialParams={{
                  initialData: drills,
                  previousScreen: 'DrillListPage',
                  previousType: DrillTypes.FRISBEE,
                }}
                listeners={({ navigation }) => ({
                  transitionStart: e => {
                    navigation.navigate = navigate;
                  },
                })}
              />
              <Stack.Screen name="DrillListPage" component={DummyScreen} />
            </Stack.Navigator>
          </NavigationContainer>
          ,
        </Provider>,
      );

      expect(getByText('3 drills available')).toBeDefined();

      await fireEvent.press(getByText('Favorites only'));

      expect(getByText('1 drill available')).toBeDefined();

      await fireEvent.press(getByTestId('validateButton'));

      expect(navigate).toBeCalledWith('DrillListPage', {
        type: DrillTypes.FRISBEE,
        currentFilters: {
          selectedFavorites: true,
          selectedLevels: [],
          selectedGoals: [],
          numberOfPlayers: undefined,
          displayedDrills: [intermediateDrill],
        },
      });
    });
  });

  describe('resets filters', () => {
    it('resets a fitler', async () => {
      const onePersonDrill = createDrill({ id: 1, minimalPlayersNumber: 1 });
      const twoPeopleDrill = createDrill({ id: 2, minimalPlayersNumber: 2 });
      const tenPeopleDrill = createDrill({ id: 3, minimalPlayersNumber: 10 });
      const drills = [onePersonDrill, twoPeopleDrill, tenPeopleDrill];
      const navigate = jest.fn();

      const DummyScreen = props => null;
      const Stack = createStackNavigator();

      const mockStore = configureMockStore()({
        favoriteDrills: [],
      });

      const { getByText, getByTestId } = render(
        <Provider store={mockStore}>
          <NavigationContainer>
            <Stack.Navigator>
              <Stack.Screen
                name="FrisbeeFilters"
                component={ConnectedFrisbeeFilters}
                initialParams={{
                  initialData: drills,
                  previousScreen: 'DrillListPage',
                  previousType: DrillTypes.FRISBEE,
                }}
                listeners={({ navigation }) => ({
                  transitionStart: e => {
                    navigation.navigate = navigate;
                  },
                })}
              />
              <Stack.Screen name="DrillListPage" component={DummyScreen} />
            </Stack.Navigator>
          </NavigationContainer>
          ,
        </Provider>,
      );

      expect(getByText('Number of players: -')).toBeDefined();
      expect(getByText('3 drills available')).toBeDefined();

      await fireEvent(getByTestId('numberOfPlayersSlider'), 'valueChange', 5);

      expect(getByText('Number of players: 5')).toBeDefined();
      expect(getByText('2 drills available')).toBeDefined();

      await fireEvent.press(getByTestId('resetButton'));

      expect(getByText('Number of players: -')).toBeDefined();
      expect(getByText('3 drills available')).toBeDefined();

      expect(navigate).toBeCalledWith('DrillListPage', {
        type: DrillTypes.FRISBEE,
        currentFilters: {
          selectedFavorites: false,
          selectedLevels: [],
          selectedGoals: [],
          numberOfPlayers: undefined,
          displayedDrills: drills,
        },
      });
    });
  });
});
