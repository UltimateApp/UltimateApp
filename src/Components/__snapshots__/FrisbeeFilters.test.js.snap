// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FrisbeeFilters /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A6A6A6",
        "marginBottom": 20,
        "paddingLeft": 20,
        "paddingTop": 10,
      }
    }
  >
    2
     drills available
  </Text>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 50,
        "paddingHorizontal": 10,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Level
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-between",
            "marginBottom": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#e5e5e5",
              "borderWidth": 1,
              "flexBasis": "30%",
              "flexGrow": 1,
              "marginRight": 5,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "textAlign": "center",
                "textTransform": "capitalize",
              }
            }
          >
            Beginner
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#e5e5e5",
              "borderWidth": 1,
              "flexBasis": "30%",
              "flexGrow": 1,
              "marginRight": 5,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "textAlign": "center",
                "textTransform": "capitalize",
              }
            }
          >
            Intermediate
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#e5e5e5",
              "borderWidth": 1,
              "flexBasis": "30%",
              "flexGrow": 1,
              "marginRight": 5,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "textAlign": "center",
                "textTransform": "capitalize",
              }
            }
          >
            Advanced
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Goals
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-between",
            "marginBottom": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexBasis": "50%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#ededed",
                "borderRadius": 3,
                "borderWidth": 0,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 0,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <Text
                style={
                  Object {
                    "color": "#43484d",
                    "fontWeight": "normal",
                    "marginLeft": 10,
                    "marginRight": 10,
                    "textTransform": "capitalize",
                  }
                }
                testID="checkboxTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                Full body
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "marginBottom": 10,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Number of players: 
        -
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "flexGrow": 1,
              "textAlign": "center",
            }
          }
        >
          1
        </Text>
        <View
          animationType="timing"
          onLayout={[Function]}
          onValueChange={[Function]}
          step={1}
          style={
            Object {
              "flexGrow": 10,
              "height": 40,
              "justifyContent": "center",
              "marginHorizontal": 10,
            }
          }
          testID="numberOfPlayersSlider"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          thumbTouchSize={
            Object {
              "height": 40,
              "width": 40,
            }
          }
          value={0}
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "backgroundColor": "#b3b3b3",
                "borderRadius": 2,
                "height": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#008577",
                "borderRadius": 2,
                "height": 0,
                "marginTop": -0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <View
            onLayout={[Function]}
            style={
              Object {
                "backgroundColor": "#008577",
                "borderRadius": 10,
                "height": 0,
                "position": "absolute",
                "top": 21,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": -0,
                  },
                ],
                "width": 0,
              }
            }
            testID="sliderThumb"
          />
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "flexGrow": 1,
              "textAlign": "center",
            }
          }
        >
          30
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
